packages/network-controller/src/NetworkController.test.ts

```diff
@@ -3722,6 +3724,103 @@ describe('NetworkController', () => {
       });
     });
   });
+
+  describe('upsertNetworkConfiguration', () => {
+    it('add a network configuration', async () => {
+      const messenger = buildMessenger();
+      await withController({ messenger }, async ({ controller }) => {
+        const rpcUrlNetwork = {
+          chainId: undefined,
+          nickname: 'RPC',
+          rpcPrefs: undefined,
+          rpcUrl: 'rpc_url',
+          ticker: 'RPC',
+        };
+
+        expect(controller.state.networkConfigurations).toStrictEqual({});
+
+        controller.upsertNetworkConfiguration(rpcUrlNetwork);
+
+        expect(
+          Object.values(controller.state.networkConfigurations),
+        ).toStrictEqual(expect.arrayContaining([rpcUrlNetwork]));
+      });
+    });
+
+    it('update a network configuration when the configuration being added has an rpcURL that matches an existing configuration', async () => {
+      await withController(
+        {
+          state: {
+            networkConfigurations: {
+              testUUID: {
+                rpcUrl: 'test_rpc_url',
+                ticker: 'old_rpc_ticker',
+                nickname: 'old_rpc_nickname',
+                rpcPrefs: { blockExplorerUrl: 'testchainscan.io' },
+                chainId: '1',
+              },
+            },
+          },
+        },
+        async ({ controller }) => {
+          controller.upsertNetworkConfiguration({
+            rpcUrl: 'test_rpc_url',
+            ticker: 'new_rpc_ticker',
+            nickname: 'new_rpc_nickname',
+            rpcPrefs: { blockExplorerUrl: 'alternativetestchainscan.io' },
+            chainId: '1',
+          });
+          expect(
+            Object.values(controller.state.networkConfigurations),
+          ).toStrictEqual([
+            expect.objectContaining({
+              rpcUrl: 'test_rpc_url',
+              nickname: 'new_rpc_nickname',
+              ticker: 'new_rpc_ticker',
+              rpcPrefs: { blockExplorerUrl: 'alternativetestchainscan.io' },
+              chainId: '1',
+            }),
+          ]);
+        },
+      );
+    });
+
+    describe('removeNetworkConfigurations', () => {
+      it('remove a network configuration', async () => {
+        const testUUID = 'testUUID';
+        await withController(
+          {
+            state: {
+              networkConfigurations: {
+                [testUUID]: {
+                  rpcUrl: 'test_rpc_url',
+                  ticker: 'old_rpc_ticker',
+                  nickname: 'old_rpc_nickname',
+                  rpcPrefs: { blockExplorerUrl: 'testchainscan.io' },
+                  chainId: '1',
+                },
+              },
+            },
+          },
+          async ({ controller }) => {
+            expect(
```

Comment:
Wondering if you need this assertion since you set the state above?

Reply:
yeah it's redundant but for me was about making sure that `withController` is doing it's job. Probably doesn't make sense to test the helper method within the test though 😅